src/main/java/cc/hyperium/gui/ChangeBackgroundGui.java

```diff
@@ -55,23 +74,35 @@ public void updateScreen() {
     @Override
     protected void keyTyped(char typedChar, int keyCode) throws IOException {
         this.downloadUrlField.textboxKeyTyped(typedChar, keyCode);
-        if (keyCode == Keyboard.KEY_ESCAPE)
+        if (keyCode == Keyboard.KEY_ESCAPE) {
             Minecraft.getMinecraft().displayGuiScreen(this.prevGui);
-        if (keyCode == Keyboard.KEY_RETURN)
+            if(isFS) {
+                Minecraft.getMinecraft().toggleFullscreen();
+            }
```

Comment:
The problem I mentioned earlier is apparent here, if you press "escape" multiple times it will switch between fullscreen and windowed mode since the value of isFS never updates.